src/app/shared/timesheet.service.spec.ts

```diff
@@ -1,12 +1,56 @@
 import { TestBed } from '@angular/core/testing';
 
+import * as _ from 'Lodash';
+
 import { TimesheetService } from './timesheet.service';
+import { Timesheet } from './timesheet.model';
+
+describe('TimesheetService - ', () => {
+
+  let service: TimesheetService;
+  const timesheet = {
+    consultant: { email: 'tester@test.com', name: 'tester', },
+    mission: { client: 'Test.com', title: 'Testin' },
+    workingDays: {
+      '0.1900': [0.5, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    },
+    invoice: null
+  };
+  const token = (mode: string) => btoa(unescape(encodeURIComponent(JSON.stringify({ mode, timesheet }))));
 
-describe('TimesheetService', () => {
   beforeEach(() => TestBed.configureTestingModule({}));
 
   it('should be created', () => {
-    const service: TimesheetService = TestBed.get(TimesheetService);
+    service = TestBed.get(TimesheetService);
     expect(service).toBeTruthy();
   });
+
+  describe('when opening a timesheet on edit mode', () => {
+    let result;
+
+    beforeEach(() => {
+      result = service.openTimesheet(token('edit'), 'edit');
+    });
+
+    it('shouldn\'t return anything if the second arg is edit', () => {
+      expect(result).toBeUndefined();
+    });
+
+    it('should return false if the second arg is not edit', () => {
+      result = service.openTimesheet(token('edit'), 'review');
```

Comment:
the opentTimesheet method does not care if the arg is "edit". It will return false if the "mode" argument does not match the "mode" stored in the token. This is the behaviour you want to describe and test